lib/Scheduler/aco.cpp

```diff
@@ -202,13 +242,13 @@ std::unique_ptr<InstSchedule> ACOScheduler::FindOneSchedule() {
   std::unique_ptr<InstSchedule> schedule =
       llvm::make_unique<InstSchedule>(machMdl_, dataDepGraph_, true);
   InstCount maxPriority = rdyLst_->MaxPriority();
-  if (maxPriority == 0)
-    maxPriority = 1; // divide by 0 is bad
+  maxPriority |= !maxPriority; // divide by 0 is bad(increase to 1 if 0)
```

Comment:
I think this is harder to understand than the if statement it replaces.

Reply:
I don't think that this is that hard to understand.  This change removes commonly executed branch from the code, which frees up a slot in the branch predictor tables.  I am open to putting a more descriptive comment if you think that would be appropriate.

Reply:
In light of the information that the compiler optimizes out the branch I have reverted these lines of code.

I know what the asm instructions do.
```asm
test Reg, Reg ; Sets the zero flag in the flags register to 1 if Reg==0, set the zero flag to 1 otherwise
cmovz Reg, 1  ; moves the immediate 1 to Reg if the zero flag == 1
```

```asm
test Reg, Reg ; Sets the zero flag in the flags register to 1 if Reg==0, set the zero flag to 1 otherwise
sete al       ; Sets al to the opposite of of the zero flag (!ZF)
or Reg, al    ; or's Reg with al
```

Reply:
Those asm instructions were rough; I didn't actually use a compiler because I was commenting on my phone. For a more accurate view into what the compiler is doing, see https://godbolt.org/z/c9YzvK (I tested it with up to the earliest versions of clang and gcc available there, and it's effectively the same code either way). So it's not exactly the same asm that I wrote (e.g. `cmov`s can't have immediate operands, it seems), but it's still branchless.